src/main/java/com/uid2/shared/auth/Roles.java

```diff
@@ -10,7 +10,7 @@
 
 public class Roles {
     public static <T extends Enum<T>> Set<T> getRoles(Class<T> type, JsonObject clientKeyJson) {
-        return getRoles(type, clientKeyJson.getJsonArray("roles"));
+        return getRoles(type, clientKeyJson.getJsonArray("roles", new JsonArray()));
```

Comment:
What's the purpose of adding this param?

Reply:
this is for backward compatitble, the new JsonArray() is the default value,  if there is no "roles" field in the JSON, this returns an empty json array. Otherwise clientKeyJson.getJsonArray("roles") will throw an exception